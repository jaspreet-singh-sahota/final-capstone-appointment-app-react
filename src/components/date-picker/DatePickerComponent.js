import React from 'react';
import PropTypes from 'prop-types';
import ReactDatePicker from 'react-datepicker';
import './styles/datepicker.css';

const DatePickerComponent = ({ selectedDate, setSelectedDate }) => (
  <ReactDatePicker
    selected={selectedDate}
    onChange={date => setSelectedDate(date)}
    minDate={new Date()}
    placeholderText="Pick a date"
    isClearable
    showTimeSelect
    timeFormat="HH:mm"
    minTime={(new Date(), 6)}
    maxTime={new Date()}
    timeIntervals={15}
    dateFormat="Pp"
    required
  />
);

DatePickerComponent.propTypes = {
  selectedDate: PropTypes.string,
  setSelectedDate: PropTypes.func.isRequired,
};

DatePickerComponent.defaultProps = {
  selectedDate: '',
};

export default DatePickerComponent;
