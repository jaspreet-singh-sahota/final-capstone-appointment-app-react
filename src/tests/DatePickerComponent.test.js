import React from 'react';
import { shallow } from 'enzyme';
import ReactDatePicker from 'react-datepicker';
import DatePickerComponent from '../components/date-picker/DatePickerComponent';

describe('DatePickerComponent', () => {
  it('should render a <ReactDatePicker /> component', () => {
    const wrapper = shallow(<DatePickerComponent />);
    expect(wrapper.find(ReactDatePicker)).toHaveLength(1);
  });

  it('matches Render snapshot', () => {
    const wrapper = shallow(<DatePickerComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
