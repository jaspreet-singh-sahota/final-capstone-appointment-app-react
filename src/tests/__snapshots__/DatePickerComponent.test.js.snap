// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerComponent matches Render snapshot 1`] = `ShallowWrapper {}`;
